### PR TITLE
null safety

### PR DESCRIPTION
**Pull request by fzyzcjy**
https://github.com/peng8350/flutter_pulltorefresh/issues/448 and https://github.com/peng8350/flutter_pulltorefresh/issues/443

### CODE REVIEW

File: lib/src/internals/indicator_wrap.dart [hunk 1]
```diff
@@ -334,30 +334,30 @@ abstract class RefreshIndicatorState<T extends RefreshIndicator>
...
-            ? SmartRefresher.ofState(context).viewportExtent
+            ? SmartRefresher.ofState(context)!.viewportExtent
             : widget.height,
         refreshStyle: widget.refreshStyle);
   }
 }
 
 abstract class LoadIndicatorState<T extends LoadIndicator> extends State<T>
-    with IndicatorStateMixin<T, LoadStatus>, LoadingProcessor {
+    with IndicatorStateMixin<T, LoadStatus?>, LoadingProcessor {
```
Comment (gedgei) [inline lib/src/internals/indicator_wrap.dart:352]: This change causes a type cast problem that prevents SmartRefresher from rendering when I set `enablePullDown` setting to true.

If you check RefreshController.headerMode and RefreshController.footerMode fields, the types used as type parameter are **not** nullable, but in IndicatorStateMixin._updateListener() there's this cast:

` ValueNotifier<V>? newMode = V == RefreshStatus
        ? refresher!.controller.headerMode as ValueNotifier<V>?
        : refresher!.controller.footerMode as ValueNotifier<V>?;`

V in this case, as seen in the line I've commented, evaluates to `LoadStatus?` and the cast fails.

I've fixed this by changing the type parameter from `LoadStatus?` to `LoadStatus` and also doing the same for `RefreshStatus?` type parameter in line 135.

Comment (fzyzcjy) [reply lib/src/internals/indicator_wrap.dart]: fixed

**Comment by fzyzcjy** [comment]
Hi, any updates?

**Comment by peng8350** [comment]
It needs time.I am not sure if this will bring breaking changes.

**Comment by werfei** [comment]
> 
> 
> It needs time.I am not sure if this will bring breaking changes.

Could you merge into a separate branch and publish a preview version?
Thank you very much.

**Comment by andoutica** [comment]
any updates on this?

**Comment by higorlapacw** [comment]
Any news?

**Comment by mikkelam** [comment]
If you guys are feeling like risking it:
```
  pull_to_refresh: 
    git:
      url: https://github.com/fzyzcjy/flutter_pulltorefresh
      ref: 0bb4f70dd7540ffe032bbc2a6994df388b4f1c39
```

Not taking any responsibility

**Comment by kuromukira** [comment]
When will this be merged and published?

**Comment by mikkelam** [comment]
Guys, give the maintainer a break. He is doing this for free, if you can't wait, start your own fork and do it yourself. 

Spamming messages here will *not* make it happen faster.

**Comment by onlyYU** [comment]
GOOD JOB !!!

**Comment by peng8350** [comment]
Thanks contribute!

**Comment by yakupbaser** [comment]
very thanks for new update :)

**Pull request merged**